core/src/sumeragi/mod.rs

```diff
@@ -85,48 +85,44 @@ impl Sumeragi {
         let network_topology = Topology::builder()
             .at_block(EmptyChainHash::default().into())
             .with_peers(configuration.trusted_peers.peers.clone())
-            .build()?;
+            .build(0)?;
 
         let sumeragi_state_machine_data = SumeragiStateMachineData {
             genesis_network,
             latest_block_hash: Hash::zeroed().typed(),
             latest_block_height: 0,
             current_topology: network_topology,
 
+            wsv: wsv.clone(),
+            transaction_cache: Vec::new(),
+
             sumeragi_thread_should_exit: false,
         };
 
-        let (incoming_message_sender, incoming_message_receiver) = std::sync::mpsc::channel();
+        let (incoming_message_sender, incoming_message_receiver) =
+            std::sync::mpsc::sync_channel(50); // TODO change to constant
```

Comment:
Or change it to a configuration variable. Or a compile-time configuration variable. The smaller the struct the more cache-friendly it is. We want to be able to profile it.

Reply:
I disagree. After I added the transaction cache this is basically a non issue. I will bump it up to a slightly higher value and then leave it because playing around with this value is not worth our time.